rafthttp/transport.go

```diff
@@ -375,6 +382,11 @@ func (t *Transport) Resume() {
 	}
 }
 
+// InitialPeerNotify returns a channel that closes when an initial
+// peer connection has been established. Use this to wait until the
+// first peer connection becomes active.
+func (t *Transport) InitialPeerNotify() <-chan struct{} { return t.initPeerNotifyCh }
```

Comment:
Even better:

InitialPeerNotify() -> ActivePeers()

ActivePeers simply calculate how many peers are active. we move the for loop thing to the raft node side in etcdserver pkg to wait for the first active peer.